ert_storage/testing/testclient.py

```diff
@@ -0,0 +1,276 @@
+import os
+import requests
+from typing import (
+    Any,
+    AsyncGenerator,
+    Generator,
+    Mapping,
+    MutableMapping,
+    Optional,
+    TYPE_CHECKING,
+    Tuple,
+    Union,
+)
+from pprint import pformat
+from contextlib import contextmanager
+from sqlalchemy.orm.session import Session
+from sqlalchemy.engine.base import Transaction
+from starlette.testclient import (
+    TestClient as StarletteTestClient,
+    ASGI2App,
+    ASGI3App,
+    Cookies,
+    Params,
+    DataType,
+    TimeOut,
+    FileType,
+)
+from sqlalchemy.orm import sessionmaker
+from graphene import Schema as GrapheneSchema
+from graphene.test import Client as GrapheneClient
+
+
+if TYPE_CHECKING:
+    from promise import Promise
+    from rx import Observable
+    from graphql.execution import ExecutionResult
+
+    GraphQLResult = Union[ExecutionResult, Observable, Promise[ExecutionResult]]
+
+
+class ClientError(RuntimeError):
+    pass
+
+
+class _TestClient:
+    __test__ = False  # Pytest should ignore this class
+
+    def __init__(
+        self,
+        app: Union[ASGI2App, ASGI3App],
+        session: sessionmaker,
+        gql_schema: GrapheneSchema,
+        base_url: str = "http://testserver",
+        raise_server_exceptions: bool = True,
+        root_path: str = "",
+    ) -> None:
+        self.http_client = StarletteTestClient(
+            app, base_url, raise_server_exceptions, root_path
+        )
+        self.gql_client = GrapheneClient(gql_schema)
+        self.session = session
+
+    def get(
+        self,
+        url: str,
+        params: Params = None,
+        headers: MutableMapping[str, str] = None,
+        cookies: Cookies = None,
+        files: FileType = None,
+        timeout: TimeOut = None,
+        allow_redirects: bool = None,
+        stream: bool = None,
+        check_status_code: Optional[int] = 200,
+    ) -> requests.Response:
+        resp = self.http_client.get(
+            url,
+            params=params,
+            headers=headers,
+            cookies=cookies,
+            files=files,
+            timeout=timeout,
+            allow_redirects=allow_redirects,
+            stream=stream,
+        )
+        _check(check_status_code, resp)
+        return resp
+
+    def post(
+        self,
+        url: str,
+        params: Params = None,
```

Comment:
Shouldn't these all be `Optional`?

Reply:
I don't really know. I've taken these verbatim from https://github.com/encode/starlette/blob/f5ecb53895b3220b78bb5ced592d7c26e8b3b6fa/starlette/testclient.py#L395-L412 and I don't think it's that smart to change them.